libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java

```diff
@@ -186,34 +224,49 @@ public void onSocialChanged(OnSocialChanged event) {
                     PushSocialPayload payload = new PushSocialPayload(mIdToken, SERVICE_TYPE_GOOGLE);
+                    AppLog.d(T.MAIN,
+                            "GOOGLE SIGNUP: onSocialChanged - error - two step authentication - dispatching "
+                            + "pushSocialLoginAction");
                     mDispatcher.dispatch(AccountActionBuilder.newPushSocialLoginAction(payload));
                     break;
                 // WordPress account exists with input email address, but not connected.
                 case USER_EXISTS:
-                    mAnalyticsListener.trackSignupSocialAccountsNeedConnecting();
-                    mAnalyticsListener.trackSignupSocialToLogin();
-                    mLoginListener.showSignupToLoginMessage();
-                    mLoginListener.loginViaSocialAccount(mGoogleEmail, mIdToken, SERVICE_TYPE_GOOGLE, true);
-                    // Kill connections with FluxC and this fragment since the flow is changing to login.
-                    mDispatcher.unregister(this);
-                    getActivity().getSupportFragmentManager().beginTransaction().remove(this).commit();
+                    AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - error - user already exists");
+                    handleUserExists();
+                    break;
+                // Too many attempts on sending SMS verification code. The user has to wait before they try again
+                case SMS_CODE_THROTTLED:
+                    AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - error - sms code throttled");
+                    showError(getString(R.string.login_error_sms_throttled));
                     break;
                 default:
-                    showErrorDialog(getString(R.string.login_error_generic));
+                    AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - error - unknown");
+                    showError(getString(R.string.login_error_generic));
                     break;
             }
             // Response does not return error when two-factor authentication is required.
-        } else if (event.requiresTwoStepAuth) {
+        } else if (event.requiresTwoStepAuth || "sms".equals(event.notificationSent)) {
+            AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - 2fa required");
             mAnalyticsListener.trackSignupSocialToLogin();
             mLoginListener.needs2faSocial(mGoogleEmail, event.userId, event.nonceAuthenticator, event.nonceBackup,
                     event.nonceSms);
-            // Kill connections with FluxC and this fragment since the flow is changing to login.
-            mDispatcher.unregister(this);
-            getActivity().getSupportFragmentManager().beginTransaction().remove(this).commit();
+            finishSignUp();
+        } else {
+            AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - shouldn't happen - google sign in success");
```

Comment:
For better differentiation between the two, we used Google *login* or *signup* rather than *signin*.  Let's update `sign in` to `signup` for consistency.